torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -858,6 +1043,21 @@ void ProcessReshapeNode(Node* n, int opset_version) {
     }
   }
 
+  if (ConstantValueMap::HasShapeValue(n->input(1)->debugName())) {
```

Comment:
should this be `else if`? Is there any way that `HasValue` and `HasShapeValue` can both return true?

Reply:
No. If the output tensor is a static shape, then `HasValue` and `HasShapeValue` are both `true`. We will do the `HasValue` logic first. Note that there are also some constraints on `input_0` to get the output shape. If such constraints not satisfied, we will evaluate `HasShapeValue` logic. So whenever the early return does not trigger, we need visit the second logic.